interfaces/builtin/udisks2.go

```diff
@@ -431,24 +425,35 @@ func (iface *udisks2Interface) AppArmorConnectedPlug(spec *apparmor.Specificatio
 }
 
 func (iface *udisks2Interface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
+	if !implicitSystemPermanentSlot(slot) {
 		spec.AddSnippet(udisks2PermanentSlotAppArmor)
 	}
 	return nil
 }
 
 func (iface *udisks2Interface) UDevPermanentSlot(spec *udev.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
-		spec.AddSnippet(udisks2PermanentSlotUDev)
-		spec.TagDevice(`SUBSYSTEM=="block"`)
-		// # This tags all USB devices, so we'll use AppArmor to mediate specific access (eg, /dev/sd* and /dev/mmcblk*)
-		spec.TagDevice(`SUBSYSTEM=="usb"`)
+	if !implicitSystemPermanentSlot(slot) {
+		var udevFile string
+		slot.Attr("udev-file", &udevFile)
```

Comment:
if we allow the slots to load an arbitrary udev snippet we probably need to deny installation by default if udev-file is set:

```
allow-installation:
   ...
   slot-attributes:
      udev-file: $MISSING
```

Reply:
Indeed - I agree - thanks for adding this.